network/servicenetwork/serialize.go

```diff
@@ -52,39 +52,88 @@ package servicenetwork
 
 import (
 	"bytes"
-	"encoding/gob"
 	"io"
 	"io/ioutil"
 
 	"github.com/ThreeDotsLabs/watermill/message"
 	"github.com/pkg/errors"
+
+	"github.com/insolar/insolar/insolar/bits"
 )
 
-// serializeMessage returns io.Reader on buffer with encoded message.Message (from watermill).
-func serializeMessage(msg *message.Message) (io.Reader, error) {
+func serializeString(buffer *bytes.Buffer, s string) {
+	// here and below: bytes.Buffer never returns errors, so do not check them
+	size := bits.NewVarSize(len(s))
+	_, _ = buffer.Write(size.Serialize())
+	_, _ = buffer.WriteString(s)
+}
+
+func deserializeString(reader io.Reader) (string, error) {
+	size, err := bits.DeserializeVarSize(reader)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to read string size")
+	}
+	result := make([]byte, size.Int())
+	if _, err = io.ReadFull(reader, result); err != nil {
+		return "", errors.Wrap(err, "failed to read string data")
+	}
+	return string(result), nil
+}
+
+func serializeMetadata(buffer *bytes.Buffer, metadata message.Metadata) {
+	keyValuePairsCount := bits.NewVarSize(len(metadata))
+	_, _ = buffer.Write(keyValuePairsCount.Serialize())
+
+	for key, value := range metadata {
```

Comment:
The key iteration order in map is guaranteed to be randomized by Golang spec, so serialized representation will not be stable. I thought of sorting the keys before serialization, however, gob (that was used previously) does not sort map keys either. So I decided to keep that behaviour unchanged for now.